CMakeLists.txt

```diff
@@ -216,6 +216,24 @@ endif()
 # allow calling ctest from the root build directory.
 enable_testing()
 
+#-----------------------------------------------------------------------------#
+# Check GCC version.
+#
+# GCC version 4.5.1 builds MiniSat incorrectly with -O2, which results in
+# incorrect answers.
+
+if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+  execute_process(
+    COMMAND gcc -dumpversion
```

Comment:
We should probably not use `gcc` here (but the cmake variable that points to the executable).

Reply:
Done